.azure/pipelines/ci.yml

```diff
@@ -29,19 +29,60 @@ variables:
 - ${{ if or(eq(variables['System.TeamProject'], 'public'), in(variables['Build.Reason'], 'PullRequest')) }}:
   - name: _BuildArgs
     value: ''
+
 jobs:
+# Build Web.JS Linux x64
+- template: jobs/default-build.yml
+  parameters:
+    codeSign: true
+    jobName: WebJS_Linux_x64_build
+    jobDisplayName: "Build: Web.JS Linux x64"
+    agentOs: Linux
+    steps:
+    - script: ./build.sh
+          --ci
+          --build-nodejs
+          --projects $(Build.SourcesDirectory)/src/Components/Web.JS/Microsoft.AspNetCore.Components.Web.JS.npmproj
+          -bl:artifacts/log/build.linux-x64.binlog
+          $(_BuildArgs)
+      displayName: Run build.sh
+    installJdk: false
+    artifacts:
+    - name: WebJS_Linux_x64_Javascript
+      path: src/Components/Web.JS/dist/
+    - name: WebJS_Linux_x64_Logs
+      path: artifacts/log/
+      publishOnError: true
+
+# Code check
 - template: jobs/default-build.yml
   parameters:
+    dependsOn:
+    - WebJS_Linux_x64_build
     jobName: Code_check
     jobDisplayName: Code check
     agentOs: Windows
     steps:
+    - task: DownloadBuildArtifacts@0
+      displayName: Download WebJS_Linux_x64_Javascript artifacts
+      inputs:
+        artifactName: WebJS_Linux_x64_Javascript
+        downloadPath: $(Build.SourcesDirectory)/src/Components/Web.JS/dist/
+        itemPattern: '**/*.js'
+    # DownloadBuildArtifacts always adds the artifact name to the path: https://github.com/Microsoft/azure-pipelines-tasks/issues/6739
+    # To workaround this, move the files manually
```

Comment:
Suggest instead using pipeline artifacts (`PublishPipelineArtifact@0` and `DownloadPipelineArtifact@2`). These are more flexible, the documented future, and don't have the same restrictions as build artifacts.

As described in the above issue, https://docs.microsoft.com/en-us/azure/devops/pipelines/artifacts/pipeline-artifacts?view=azure-devops&tabs=yaml, and https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/utility/download-pipeline-artifact?view=azure-devops, the `DownloadPipelineArtifact@2` only adds the artifact name to the given destination folder when downloading from multiple containers.

Reply:
Ah didn't know there was this alternative. Will look into using it.